frappe/public/js/frappe/form/controls/base_control.js

```diff
@@ -183,7 +183,15 @@ frappe.ui.form.Control = class BaseControl {
 						// onchange event specified in df
 						let set = (me.df.change || me.df.onchange).apply(me, [e]);
 						me.set_invalid && me.set_invalid();
-						return set;
+
+						// change/onchange functions can also return Promise
+						if (set && set.then) {
+							return set.then(function (val) {
+								 return val;
+							});
+						} else {
+							return set;
+						}
```

Comment:
no sure if this is really needed as we just use change/onchange function for executing stuff